src/components/Swap/index.tsx

```diff
@@ -61,10 +66,17 @@ export default function Swap(props: SwapProps) {
 
   const focused = useHasFocus(wrapper)
 
+  // FIXME(kristiehuang): clarify when the integrator flow should open
+  const shouldOpenIntegratorFlow = Boolean(props.provider) || Boolean(props.onClickConnectWallet)
+
   return (
     <>
       <Header title={<Trans>Swap</Trans>}>
-        <Wallet disabled={!active || Boolean(account)} onClick={props.onConnectWallet} />
+        <ConnectWallet
```

Comment:
This should still be the `Wallet`, not `ConnectWallet`. The connecting logic/components should be encompassed in the `Wallet`, not vice versa.